gdnative-core/src/core_types/color.rs

```diff
@@ -41,6 +41,88 @@ impl Color {
         Color::from_sys(unsafe { (get_api().godot_color_from_hsv)(color.sys(), h, s, v, a) })
     }
 
+    /// Parses from a HTML color code, or `None` on parse error.
+    ///
+    /// ```
+    /// use gdnative::prelude::Color;
+    ///
+    /// // Each of the following creates the same color RGBA(178, 217, 10, 255).
+    /// let c1 = Color::from_html("#9eb2d90a"); // ARGB format with "#".
+    /// let c2 = Color::from_html("9eb2d90a");  // ARGB format.
+    /// let c3 = Color::from_html("#b2d90a");   // RGB format with "#".
+    /// let c4 = Color::from_html("b2d90a");    // RGB format.
+    ///
+    /// let expected = Color::from_rgba_u8(178, 217, 10, 158);
+    /// assert_eq!(c1, Some(expected));
+    /// assert_eq!(c2, Some(expected));
+    ///
+    /// let expected = Color::from_rgba_u8(178, 217, 10, 255);
+    /// assert_eq!(c3, Some(expected));
+    /// assert_eq!(c4, Some(expected));
+    /// ```
+    ///
+    /// See also the corresponding [GDScript method](https://docs.godotengine.org/en/stable/classes/class_color.html#class-color-method-color).
+    #[inline]
+    pub fn from_html(mut argb_or_rgb: &str) -> Option<Self> {
```

Comment:
Does this need to be mutable?  I can't imagine why an API user would want their color code mutated by this method.  and it seems like your just mutating it like some internal variable.

Reply:
This `mut` only refers to the parameter, not the instance. It means that I can reassign the `argb_or_rgb` _inside_ the method.

The method `from_html` is a "constructor" (returning `Self`), so it can be called like this:
```rs
let color = Color::from_html("#9eb2d90a");
```

Since it has no `&mut self` parameter, the method cannot modify any existing instance.
Thanks for the extra pair of eyes! 😉

Reply:
Sure, but I don't see why the affordance of reusing that variable name in the function should be exposed to API users when you can just use a different name internally instead

Reply:
The `mut` binding is not part of the interface, so there's no reason why I would complicate my code.

See https://godot-rust.github.io/docs/gdnative/core_types/struct.Color.html#method.from_html